### PR TITLE
Pass ctx to constructor

### PR DESCRIPTION
**Pull request by ilya-hontarau**
New() makes a few remote calls, so I think it would be great to pass context to this function.
Also in my opinion retrying shouldn't be a part of a constructor(at least if don't specify any configs parameters).

So I made a breaking change(is it okay?), added context to New(), remove retrying, added ErrPing error on failed to ping.

Let me know what do you think.

### CODE REVIEW

File: pgmq.go [hunk 1]
```diff
@@ -4,18 +4,19 @@ import (
...
 	"github.com/jackc/pgx/v5/pgconn"
 	"github.com/jackc/pgx/v5/pgxpool"
 )
 
 const vtDefault = 30
 
-var ErrNoRows = errors.New("pgmq: no rows in result set")
+var (
+	ErrNoRows = errors.New("pgmq: no rows in result set")
+	ErrPing   = errors.New("failed to ping db")
```
Comment (craigpastro) [inline pgmq.go:18]: Maybe for consistency?
```suggestion
	ErrPing   = errors.New("pgmq: failed to ping db")
```

**Comment by craigpastro** [comment]
Hi @ilya-hontarau! Thank you for your contribution! I think it is a good idea. Tests are failing because there was a breaking change in pgmq recently. I just updated so it should work now if you rebase. I left a few stylistic comments. Thanks again!

File: pgmq_test.go [hunk 2]
```diff
@@ -8,12 +8,14 @@ import (
...
 	"time"
 
-	"github.com/craigpastro/pgmq-go/mocks"
+	"github.com/craigpastro/retrier"
 	"github.com/jackc/pgx/v5/pgconn"
 	"github.com/stretchr/testify/require"
 	"github.com/testcontainers/testcontainers-go"
 	"github.com/testcontainers/testcontainers-go/wait"
 	"go.uber.org/mock/gomock"
+
```
[the remark this answers is not in the document]
Comment (ilya-hontarau) [reply pgmq_test.go]: will be great to use imports formatters

[the remark this answers is not in the document]
Comment (craigpastro) [reply pgmq_test.go]: Yeah, I had thought that the `goimports` linter would have caught this, but it doesn't complain about newlines. Maybe `gci` works better? Do you know any alternatives?

[the remark this answers is not in the document]
Comment (craigpastro) [reply pgmq_test.go]: It's weird that goimports is not complaining in ci either. Something seems to be not working there. Oh well.

**Review by craigpastro — APPROVED** [review]
LGTM. Thanks!

File: pgmq_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilya-hontarau) [reply pgmq_test.go]: @craigpastro actually I like to split imports in 3 group:
* std
* external imports
* internal imports

I tried before GCI, but it still couldnt do this(maybe now they can?).
Also https://github.com/incu6us/goimports-reviser works fine, but it isn't a part of golangci lint:(

[the remark this answers is not in the document]
Comment (craigpastro) [reply pgmq_test.go]: I am pretty happy with
- std
- everything else

I am not sure how gci was working in the past but it seems that they can do what you like now by specifying something like:
```
linters-settings:
  gci:
    sections:
      - standard
      - default
      - prefix(github.com/craigpastro/pgmq-go)
```

By the way, I think the issue with ci is that the linter config contained:
```
issues:
  fix: true
```
If I remove it then it fails when I expect it to. 🤷

**Pull request merged**